drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -0,0 +1,274 @@
+    def perform
+      # warning- this destroys all referrals regardless of data source
+      referral_class.destroy_all if clobber
+
+      import_enrollment_records
+      import_referral_records
+      import_referral_posting_records
+      import_referral_household_members_records
+      assign_unit_occupancies
+    end
+
+    POSTINGS_FILENAME = 'ReferralPostings.csv'.freeze
+    HOUSEHOLD_MEMBERS_FILENAME = 'ReferralHouseholdMembers.csv'.freeze
+
+    def runnable?
+      super &&
+      reader.file_present?(POSTINGS_FILENAME) &&
+      reader.file_present?(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    protected
+
+    def referral_class
+      HmisExternalApis::AcHmis::Referral
+    end
+
+    def supports_upsert?
+      true
+    end
+
+    #
+    # Data Access
+    #
+    def posting_rows
+      reader.rows(POSTINGS_FILENAME)
+    end
+
+    def household_member_rows
+      reader.rows(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    #
+    # Importers
+    #
+    def import_enrollment_records
+      # and easily do bulk upsert here due to dependent WIP records
+      without_paper_trail do
+        build_enrollment_rows.each(&:save!)
+      end
+    end
+
+    def import_referral_records
+      ar_import(
+        referral_class,
+        build_referral_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_posting_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralPosting,
+        build_posting_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_household_members_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralHouseholdMember,
+        build_household_member_records,
+        on_duplicate_key_update: { conflict_target: [:client_id, :referral_id], columns: :all },
+      )
+    end
+
+    #
+    # Builders
+    #
+
+    # create wip enrollments for accepted pending postings
+    def build_enrollment_rows
+      accepted_pending_status = HmisExternalApis::AcHmis::ReferralPosting.statuses.fetch('accepted_pending_status')
+      enrollment_coc_by_project_id = Hmis::Hud::ProjectCoc
+        .where(data_source: data_source)
+        .pluck(:project_id, :coc_code)
+        .to_h
+
+      posting_rows.flat_map do |posting_row|
+        next unless posting_status(posting_row) == accepted_pending_status
+
+        project_id = row_value(posting_row, field: 'PROGRAM_ID')
+        referral_id = row_value(posting_row, field: 'REFERRAL_ID')
+        household_id = Hmis::Hud::Base.generate_uuid
+        household_member_rows_by_referral(referral_id).map do |member_row|
+          entry_date = parse_date(row_value(posting_row, field: 'REFERRAL_DATE'))
+          mci_id = row_value(member_row, field: 'MCI_ID')
+          enrollment = Hmis::Hud::Enrollment.new(default_attrs)
+          enrollment.attributes = {
+            personal_id: mci_id,
```

Comment:
@ttoomey I missed this the first time, it's not quite right. The Personal ID of the enrollment needs to match `Client.PersonalID` of the client record. So really its `Hmis::Hud::Client.find(client_pk_by_mci_id(mci_id)).personal_id`. Not sure if there is already a mapping like that (mci => personal id). Can you update?

Reply:
Capturing discussion with @gigxz on this: The Personal ID the "MCI unique ID", not the "MCI ID". The column in the sheet is the "MCI ID" so the mapping isn't direct, need to go through external IDs instead.
* build mapping from MCI ID using external ids
* fix other importers that map MCI ID => Personal ID

Reply:
Corrected this, looks like the referrals were the only impart impacted (the client address / contact points use personal ids)